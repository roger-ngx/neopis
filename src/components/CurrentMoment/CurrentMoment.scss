.current_moment{
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

.current_icon{
  width: 17px;
  height: 17px;
  object-fit: contain;
}

.current_date{
  font-family: HiraKakuProN-W3;
  font-size: 16px;
  letter-spacing: -0.5px;
  text-align: center;
  color: #ffffff;
  margin-left: 10px;
}

.current_time{
  font-family: HiraKakuProN-W6;
  font-size: 19px;
  letter-spacing: -0.6px;
  text-align: center;
  color: #ffffff;
  margin-left: 10px;
}