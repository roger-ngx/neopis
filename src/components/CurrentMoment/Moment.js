import React, { Component } from 'react';
import * as moment from 'moment';
import 'moment/locale/ko';

export function withMoment(Current) {
    return class Moment extends Component {

        constructor(props) {
            super(props);
            moment.locale('kr');

            this.state = {
                date: '',
                time: ''
            }
        }

        componentDidMount() {
            setInterval(() => {
                const [date, time] = moment().format('YYYY / MM / DD / dddd,HH:mm:ss').split(',');
                this.setState({ date, time })
            }, 1000);
        }

        render() {
            return <Current {...this.state} />
        }
    }
}