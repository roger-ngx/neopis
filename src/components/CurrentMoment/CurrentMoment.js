import React from 'react';
import { withMoment } from './Moment';
import { PropTypes } from 'prop-types';

import './CurrentMoment.scss';
import image from '../../assets/images/time.svg'

export const CurrentMoment_ = (props) => <div className='current_moment'>
  <img className='current_icon' src={image} alt='time icon' />
  <div className='current_date'>
    {props.date}
  </div>
  <div className='current_time'>
    {props.time}
  </div>
</div>

CurrentMoment_.propTypes = {
  date: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired
}

export default withMoment(CurrentMoment_);