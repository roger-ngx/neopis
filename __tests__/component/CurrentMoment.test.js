import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {CurrentMoment_} from '../../src/components/CurrentMoment/CurrentMoment';
import { withMoment } from '../../src/components/CurrentMoment/Moment';

Enzyme.configure({ adapter: new Adapter() })

describe('Current Moment component test', () => {
  it('should render datetime', () => {
    const CurrentMoment = withMoment(CurrentMoment_);

    const wrapper = shallow(<CurrentMoment />)

    expect(wrapper.first().is(CurrentMoment_)).toBe(true);
    
    const instance = wrapper.instance();

    instance.setState({date:'2132', time: '00000'});

    expect(wrapper.first().prop('time')).toEqual('00000');
  })

  it('test rendering', () => {
    const tree = renderer.create(<CurrentMoment_ />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CurrentMoment_ />, div);
  })
})