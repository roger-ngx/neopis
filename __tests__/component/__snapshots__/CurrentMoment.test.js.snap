// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current Moment component test test rendering 1`] = `
<div
  className="current_moment"
>
  <img
    alt="time icon"
    className="current_icon"
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <div
    className="current_date"
  />
  <div
    className="current_time"
  />
</div>
`;
